Makefile.am

```diff
@@ -40,15 +40,15 @@ test: check-recursive
 
 # Target to make autogenerated files.
 checkprograms :
-	(cd src ; $(MAKE) libsndfile.la checkprograms)
-	(cd tests ; $(MAKE) checkprograms)
+	$(MAKE) -C src libsndfile.la checkprograms
+	$(MAKE) -C tests checkprograms
```

Comment:
1. you don't check whether cd fails
2. the (...) construct opens a subshell, which should generally be avoided
3. `make -C` fulfills both of the previosu requirements